### ISSUE TITLE
There is no default maxheight in Israel

### ISSUE THREAD

**Issue by zstadler** [streetcomplete/StreetComplete#5458]
StreetComplete enters an invalid  `maxheight=default` for ways in Israel. There is no default height in Israel. 
Moreover, any effort to remove the erroneous tags is doomed to fail, as new StreetComplete edits keep adding them.

**How to Reproduce**
Open the maxheight quest and enter `maxheight=default` for a way in Israel.
For example, https://www.openstreetmap.org/changeset/138343697

**Expected Behavior**
The `default` value should not be an option for maxheight in Israel.

**Versions affected**
All

**Comment by zstadler** [comment]
> Are you sure that Israel has no legal size limit for vehicles driving on public roads? Or are you claiming that it has limits on say width, it has no limit on vehicle height?

There are limitations on vehicles, depending on their size. There is also a procedure for police approval for handling oversized transportation. However, this is not a limitation on the roads.

> Also, in https://www.openstreetmap.org/way/881547522/history you mention
> 
> > there is always explicit sign with numbers.

This is not my changeset and not my comment, but this is correct. There should be a maxheight sign for every height obstacle on an Israeli road.

> in such case if user selected "there is no sign" they were wrong, but SC requires no changes.
> (but I would expect this to be not true and at least some places being unsigned - maybe sign fallen off or was stolen)

Regardless of the locality:
- If there is a maxheight sign, and the user ignores it, then the user is indeed wrong. Am I correct to assume that SC requests a confirmation for this choice?
- On the other hand, if there is no maxheight sign, for any reason, and the user chooses a "there is no sign" in SC, then the user is not wrong. 

A missing maxheight sign in Israel does not mean that "the physical height is sufficient for the country specific height limit of general traffic", as the [wiki explains `maxheight=default`](https://wiki.openstreetmap.org/wiki/Key:maxheight#Non-numerical_values). It means that sign could have been anything, and the _physical height_ can be of any value. Therefore, a `maxheight=default` tag on any road in Israel is incorrect and misleading.

If a SC user selects a "there is no sign" in Israel, then [according to the wiki](https://wiki.openstreetmap.org/wiki/Key:maxheight#Non-numerical_values), SC should use the `maxheight=no_sign` tag, rather than `maxheight=default`. SC is responsible for the wording of the options presented to its users and their interpretation as specific OSM tagging to perform.

IMO, an explicit `maxheight=no_sign` tag could be a great way for the authorities and citizens to use SC-driven surveys to identify and fix cases of missing maxheight signs.

**Comment by matkoniecz** [comment]
> This is not my changeset and not my comment

ops, sorry

> A missing maxheight sign in Israel does not mean that "the physical height is sufficient for the country specific height limit of general traffic"

I agree! That is why StreetComplete asks whether 

> Is there easily enough space for even the tallest trucks (at least about 4.5 meters / 15 feet) to pass through?
> - Tall trucks can pass through
> - Tall trucks don’t fit

(`quest_maxheight_answer_noSign_question` `quest_maxheight_answer_noSign_question_yes` `quest_maxheight_answer_noSign_question_no`)

If first option is taken then `maxheight=default` is tagged and otherwise `maxheight=below_default` is tagged, to distinguish between cases where there is no heigh limit applying to normal vehicles (though oversized loads may be affected) and between cases where there is limit applying to normally sized vehicles but without sign.

> If a SC user selects a "there is no sign" in Israel, then [according to the wiki](https://wiki.openstreetmap.org/wiki/Key:maxheight#Non-numerical_values), SC should use the maxheight=no_sign tag

But SC asks more, to be more specific.

**Comment by zstadler** [comment]
Since the notion of a "default height" has no meaning in Israel, as explained in length above, both `maxheight=default` and `maxheight=below_default` are inapplicable. As a result, removal of the `maxheight=default` tags in Israel is bound to continue.

Please use the meaningful and useful `maxheight=no_sign` tag in Israel when a user selects "there is no sign", as I've suggested.

**Comment by matkoniecz** [comment]
> Since the notion of a "default height" has no meaning in Israel

Why you think so? You mention "There is also a procedure for police approval for handling oversized transportation." so it means that there is size limit for vehicles below which they are not an oversized vehicle (this typically is the same size as typical HGV/TIR/large truck operating on roads).

**Comment by matkoniecz** [comment]
@yrtimiD also feel free to answer that question, here or at https://www.openstreetmap.org/changeset/146652481

**Comment by yrtimiD** [comment]
I was about to write exactly the same words, to me the `maxheight=no_sign` option seems to only usable option in Israel realities. 
Actually, the "default" value might fit only in very few cases where space under the bridge is enormous like [here](https://www.google.com/maps/@31.803011,35.1826809,3a,75y,74.84h,100.6t/data=!3m7!1e1!3m5!1sb_JScYLmyVjaCE_ZhGqwBQ!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fpanoid%3Db_JScYLmyVjaCE_ZhGqwBQ%26cb_client%3Dsearch.revgeo_and_fetch.gps%26w%3D96%26h%3D64%26yaw%3D314.40454%26pitch%3D0%26thumbfov%3D100!7i13312!8i6656?entry=ttu)
![image](https://github.com/streetcomplete/StreetComplete/assets/816805/de5b6a38-458c-4294-86c8-a5110f178fae)

There is another issue, I noticed that most "default" reports were in places where there are proper and clear signs. (guess people reported while driving and just didn't notice the sign). As result, this quest was hidden from other users without chance to fix. As a solution, if we'll write "no_sign" then we can re-activate these quests later and ask if "still no sign?".

**Comment by yrtimiD** [comment]
I was about to write exactly the same words, to me the `maxheight=no_sign` option seems to only usable option in Israel realities. 
Actually, the "default" value might fit only in very few cases where space under the bridge is enormous like [here](https://www.google.com/maps/@31.803011,35.1826809,3a,75y,74.84h,100.6t/data=!3m7!1e1!3m5!1sb_JScYLmyVjaCE_ZhGqwBQ!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fpanoid%3Db_JScYLmyVjaCE_ZhGqwBQ%26cb_client%3Dsearch.revgeo_and_fetch.gps%26w%3D96%26h%3D64%26yaw%3D314.40454%26pitch%3D0%26thumbfov%3D100!7i13312!8i6656?entry=ttu)
![image](https://github.com/streetcomplete/StreetComplete/assets/816805/de5b6a38-458c-4294-86c8-a5110f178fae)
And even here, only few kilometers ahead before entering this road there are plenty of signs:
![image](https://github.com/streetcomplete/StreetComplete/assets/816805/d66e50cb-17ef-4d39-8236-79fb52899c9b)

There is another issue, I noticed that most "default" reports was in places where there are proper and clear signs. (guess people reported while driving and just didn't notice the sign). As result, this quest was hidden from other users without chance to fix.

**Comment by matkoniecz** [comment]
Is `default` wrong for road like https://www.google.pl/maps/@50.0870948,19.8056609,3a,60y,271.51h,88.27t/data=!3m6!1e1!3m4!1slouelE-fm_uZ6C4lv-LTaA!2e0!7i16384!8i8192?entry=ttu if it would be located in Israel? If wrong - why?

![screen02](https://github.com/streetcomplete/StreetComplete/assets/899988/73f8bb4f-b346-42c0-a0df-f8c203c3ace3)


> I noticed that most "default" reports was in places where there are proper and clear signs

In such case: please comment on changeset that people made and ask them to be more careful

**Comment by yrtimiD** [comment]
> Or is such situation impossible for some reason? (why?)

From my experience - in Israel there will be a clear sign on the bridge for each lane. And even if there is declared global default - only very special people will know it (unlike max speed, for example, which is well known to every driver)

> In such case: please comment on changeset that people made and ask them to be more careful

Of course, I send them all messages with explanation, but it feels like these people are not "pure" OSM editors and contribute only via SC, so they don't know what are these "default" or "below_default" means at all. I'd just prefer to avoid entering incorrect values in the first place.

**Comment by zstadler** [comment]
Since there is a requirement for maxheight signs in Israel, the lack of a sign cannot be interpreted as "the physical height is sufficient for the country specific height limit of general traffic". It should be interpreted as "there is no sign".

@matkoniecz,
I'm having a hard time understanding the purpose of this intense discussion. 
Are you trying to convince the Israeli mappers that `maxheight=default` is better than `maxheight=no_sign` or no `maxheight` tag at all?
Is your goal to avoid a change in SC rather than increase its value in Israel?

**Comment by matkoniecz** [comment]
Do you have an Android phone? Have you seen how this question is presented there? Can you look at it in app?

> the lack of a sign cannot be interpreted as "the physical height is sufficient for the country specific height limit of general traffic"

it is **not** interpreted this way by StreetComplete.

After user marks "no sign" answer, then StreetComplete additionally asks whether 

> Is there easily enough space for even the tallest trucks (at least about 4.5 meters / 15 feet) to pass through?
> - Tall trucks can pass through
> - Tall trucks don’t fit

`maxheight=default` is used for "there is no sign and there is enough space for even the tallest trucks (at least about 4.5 meters / 15 feet) to pass through"

`maxheight=below_default` is used for "there is no sign and there is not enough space for tall trucks"

It is worth distinguishing this cases.

>  I'm having a hard time understanding the purpose of this intense discussion.

I am trying to understand root of the reported problem so it can be fixed properly.

>  Are you trying to convince the Israeli mappers that maxheight=default is better than maxheight=no_sign or no maxheight tag at all?

In cases where `maxheight=default` is applicable, yes, I am trying to do this and it provides more info than `maxheight=no_sign`.

But also willing to be convinced (see for example sidewalks of cycleways discussion).

But with even higher priority, I want to understand situation, and what specifically went wrong so it can be solved.

Changing app behaviour without understanding what is being fixed is asking for trouble, in many ways.

**Comment by yrtimiD** [comment]
I do have android with SC and this quest is one of my favorites.
The change we want is simple - if country is Israel, just stop on "no sign" and do not ask the extra "Is there easily enough space for even the tallest trucks". In our case the "no_sign" is good enough.

**Comment by mnalis** [comment]
> The change we want is simple - if country is Israel, just stop on "no sign" and do not ask the extra "Is there easily enough space for even the tallest trucks".

I could be wrong (I have only written few SC quests), but I think that SC quests currently does not have an ability to modify quests answers depending on the country. There is only ability to **disable showing the quest completely in some country**, but that is obviously sub-optimal. (although I do seem to recall discussion about allowing the code to inspect the country, I cannot find it right now, but AFAIR it was ultimately rejected. Perhaps someone would like to get consensus on that and implement code which can then be reused on quests?) 

So ideally, the solution chosen would be applicable and useful in the whole world (instead of disabling maxheight quest in Israel). 

- Perhaps extra answer  in addition to `default` and `below_default` which would tag [maxheight](https://wiki.openstreetmap.org/wiki/Key:maxheight)=[no_sign](https://wiki.openstreetmap.org/wiki/Tag:maxheight=no_sign) named something like _"there is no sign, and I don't know the details why"_ or similar might be suitable? Although that value, while documented, has a very low usage.

- Or document on the wiki what that `default` value might mean in countries which do not legally prescribe the limit? 

- Or a bot which replaces undefined maxheight values in Israel with `fixme=needs resurvey to tag correct maxheight`?

- Or something else? Is there Israeli community discussion on that subject, can someone link to it (or create new discussion if it was not previously discussed)? 

- As said, there is always an option to completely disable that quest if  the that country's community consensus  is that such quest is doing more damage than good in some their country; but some other solution is usually preferred.

**Comment by westnordost** [comment]
> I could be wrong (I have only written few SC quests), but I think that SC quests currently does not have an ability to modify quests answers depending on the country

You are wrong, it is possible to distinguish countries in the form.

**Comment by westnordost** [comment]
Hey, don't cheer me, I did not offer a solution, I just read the first sentence of mnalis last answer and thought I'd correct this. In fact, I didn't follow this ticket otherwise as @matkoniecz was already in dialogue with @zstadler (I don't need to be involved in every ticket)

**Comment by yrtimiD** [comment]
we are cheering to the fact that country checking is already exists, so the solution should be pretty easy to implement :)

**Comment by yrtimiD** [comment]
@mnalis ,

> * Perhaps extra answer  in addition to `default` and `below_default` which would tag [maxheight](https://wiki.openstreetmap.org/wiki/Key:maxheight)=[no_sign]
(https://wiki.openstreetmap.org/wiki/Tag:maxheight=no_sign) named something like _"there is no sign, and I don't know the details why"_ or similar might be suitable? Although that value, while documented, has a very low usage.

In current implementation the "no sign" option is slightly misleading as at the end `maxheigh=no_sign` will not be added. Maybe your suggestion of 3rd option "just no sign" will help, but it still allows selecting meaningless option (which can be prevented)

> * Or document on the wiki what that `default` value might mean in countries which do not legally prescribe the limit?

Unfortunately, in our case there is no clear definition of what it might mean. the no_sign is way cleaner.

> * Or a bot which replaces undefined maxheight values in Israel with `fixme=needs resurvey to tag correct maxheight`?

That's actually the main problem - at the moment somebody adds `maxheight=default` - all validators including SC assume positive input and hide problematic places!

> * Or something else? Is there Israeli community discussion on that subject, can someone link to it (or create new discussion if it was not previously discussed)?

We did discussion in the telegram, mostly in hebrew.

> * As said, there is always an option to completely disable that quest if  the that country's community consensus  is that such quest is doing more damage than good in some their country; but some other solution is usually preferred.

No-no, currently quest is very useful and contributes significantly more than harm, no reason to disable it.

**Comment by westnordost** [comment]
`maxheight:signed=no` is added by StreetComplete when the user adds a physical maxheight (by measuring it, e.g. with the StreetMeasure app) in the `AddMaxPhysicalHeight` quest, which is asked for roads below bridges that have `maxheight = below_default or source:maxheight ~ ".*estimat.*" or (maxheight:signed = no and !maxheight)` tagged.

@matkoniecz could you summarize the outcome of this discussion? Would you say it should be implemented as suggested? Which tag to use?

**Comment by matkoniecz** [comment]
As I understand, according to @yrtimiD and @zstadler `maxheight=default` / `maxheight=below_default` is unwanted in Israel and that `maxheight=no_sign` is preferred.

I have not managed to figure out why.

**Comment by zstadler** [comment]
I don't know how else can we say that wasn't already said. Nevertheless, I'll rephrase yet again:

Israel has no default height in terms signage. Therefore tagged a road with `maxheight=default` / `maxheight=below_default` make no sense. These tags will be removed periodically. We kindly ask that SC refrains from creating them.

OSM tagging should stand by itself, regardless of the site/app used for that tagging. Currently, SC asks a question that is irrelevant for Israel and uses this respond to choose between two irrelevant tags.

_______
@westnordost As far as I can see, SC Does not add a `maxheight:signed=no` tag. For example, https://www.openstreetmap.org/way/206995509/history has this SC edit:

![image](https://github.com/streetcomplete/StreetComplete/assets/1304610/3c5994b5-76e8-467e-8a4b-e83b75dc9ec4)

**Comment by matkoniecz** [comment]
> Israel has no default height in terms signage. 

That is exactly the same situation as elsewhere. But I would expect that, also like elsewhere, Israel has standard maximum height for regular vehicles (trucks will be typically having this size).

And it is useful to distinguish between "unsigned, there is space for full sized vehicles" and "unsigned, full sized truck will not fit".

And if for one reason or another making this distinction is impossible/unwanted then it applies not only in Israel.

**Comment by westnordost** [comment]
Uhm, now I am confused, too. As @matkoniecz wrote, you won't find "default max height" signs anywhere, so I fail to understand in what way Israel differs from elsewhere. (Someone wrote that all maxheights are always signed in Israel? Well, then, allowing users to tag `maxheight=no_sign` equally wouldn't make sense and any non-number would be a mistake - a mistake committed by the user.)

> @westnordost As far as I can see, SC Does not add a maxheight:signed=no tag. 

Well, believe me, it does. Your link leads to a changeset that was done in the frame of the `AddMaxHeight` quest, not the `AddMaxPhysicalHeight` quest.

**Comment by matkoniecz** [comment]
> I fail to understand in what way Israel differs from elsewhere

I suspect that situation is the same as elsewhere - but there may be local preference to one tagging over another. Or maybe wiki page in Hebrew is badly translated? We have seen this type of confusion before (commas vs semicolons in opening hours where misunderstanding was result of badly translated German wiki page).

**Comment by yrtimiD** [comment]
> And it is useful to distinguish between "unsigned, there is space for full sized vehicles" and "unsigned, full sized truck will not fit".

Regardless to the country, I'm as a driver without truck license will prefer to not answer a question about "full sized vehicle" at all. I really don't know how high these vehicles are and in most cases can't even say if a bridge is high enough. Giving such "unprofessional" answer might be very problematic, and from UX point of view user is kind of forced to choose one of options instead of just finishing with "no sign". (I observe this on many changes in Israel done with SC. People selected absolutely random value where maxheight sign was absent or person just missed it)

**Comment by westnordost** [comment]
I see, but then this is not specific to Israel. I.e. this should be discussed in the community forums or mailing lists.

I'd be fine if SC always tagged something along the lines of `maxspeed=no_sign` everywhere if there is no sign, to be honest, I'd be glad to, because you *do* have a point and I pretty much agree on it, actually. But then, this should be done everywhere, not just Israel.

Question is, why was it implemented with default/below_default in the first place? Maybe back when it was implemented, only default/below_default were considered valid?

This is the PR that added the quest back then: https://github.com/streetcomplete/StreetComplete/pull/960
I didn't look through it yet if there was any discussion about that.

**Comment by FloEdelmann** [comment]
I read through it and these are the relevant comments/threads:

* #960 (PR description)
* https://github.com/streetcomplete/StreetComplete/pull/960#discussion_r173305043
* https://github.com/streetcomplete/StreetComplete/pull/960#discussion_r173318315
* https://github.com/streetcomplete/StreetComplete/pull/960#discussion_r199282918
* https://github.com/streetcomplete/StreetComplete/pull/960#issuecomment-372014605

**Comment by matkoniecz** [comment]
> ...but tag maxheight=default if the road is a highway=motorway or motorway_link or

I would not do this, there are some really silly cases of road technically being motorway but... If such assumption can be reliably made - then it can be applied in processing data.

**Comment by mnalis** [comment]
> So, StreetComplete could tag `maxheight:signed=no` in case the user answers that no `maxheight` is signed.

... Thus replacing  `maxheight=default` and `maxheight=below_default` answers, yes? Makes sense to me, as the those two sub-answers were complicated for users to answer correctly anyway (especially the ones who do not drive trucks nor interact with them often), thus I would too prefer to just have one simple answer saying it is not signed.

Users who really care and wish to add extra details to those unsigned maxheights would use `AddMaxPhysicalHeight` quest to fill in extra details more precisely (with all the extra effort involved, of course).

> >    ...but tag `maxheight=default` if the road is a `highway=motorway` or `motorway_link` or

>   I would not do this, there are some really silly cases of road technically being motorway but...

I agree, better not to add `maxheight=default` there, as it would be an assumption and not measured/confirmed by user (and as _assumption_, it can better be made in final data consumer with any special casing they might want to do). Skipping the quest on motorways is fine, of course.

**Comment by Helium314** [comment]
I actually find the `below_default` answer quite useful for (in may area) frequently unsigned ways, e.g. building passages. `AddMaxPhysicalHeight` is not an option, as I neither want to carry a long pole for measuring, nor use a different phone just for arcore.

**Comment by westnordost** [comment]
The usefulness of `below_default` and `default` over `no_sign` is undisputed.

If you answer `below_default`, the `AddMaxPhysicalHeight` quest will pop up just as much as it would when you answer that it is `maxheight:signed=no`. (And if your phone does not support ARCore, you'll have that quest disabled, anyway)

**Comment by FloEdelmann** [comment]
When answering "there is no sign", currently a dialog pops up that asks whether tall trucks fit under the bridge. In that dialog, we could simply add a third option "I can't tell" that tags `maxheight:signed = no`, while the other two options tagging `maxheight = below_default` and `maxheight = default` could stay there for people who know the difference.

**Comment by westnordost** [comment]
That crossed my mind, too, but I always found this dialog unnecessarily complex for the mapper. And it doesn't really lift that much of a burden from the mapper, as then he'd need to decide if he's sure about it (and how sure is sure?).

**Comment by matkoniecz** [comment]
It kind of would be nice to be able to tag blatant cases (2m high building passage, 120h clearance under a bridge) but I can live with eliminating it altogether.

**Comment by mnalis** [comment]
> I actually find the below_default answer quite useful for (in may area) frequently unsigned ways, e.g. building passages. AddMaxPhysicalHeight is not an option, as I neither want to carry a long pole for measuring, nor use a different phone just for arcore.

Understood. On the other hand I for example have no idea whatsoever how high "tall"  truck as opposed to "short" truck is. I am also quite bad at estimating how wide/high something is, and whether something else (which I don't even see at that time!) is wider/taller than that :shrug: 

> Anyway, I am not set on this, I am just voicing my opinion that I'd like to use this to eliminate that dialog altogether.

Perhaps solution to address the both issues might be:

- have only one `maxheight:signed=no`  answer in `AddMaxHeight`  quest
- move existing answers `maxheight=default` and `maxheight=below_default` from it to `AddMaxPhysicalHeight` quest (as   `other answers`)

It would:
- keep `AddMaxHeight` simple for users who just want to record what they see, without obtaining expertise (or guessing) how tall some truck might be, thus keeping with _"👨‍💻 Users are no experts"_ and _"🐿️ Easy answer"_ tenets
- still allow users who wish to provide more data (but can't measure exact heights) to answer if it is `default` or `below_default` legal limit (albeit with one more click) 
- solve this specific issue for Israel and similar countries which do not have "default" legal limit as defined by the [wiki](https://wiki.openstreetmap.org/wiki/Key:maxheight)

**Comment by westnordost** [comment]
So, in short

- if user answers "has no sign" in `AddMaxHeight`-quest simply add `maxheight:signed=no`

- for `AddMaxPhysicalHeight`-quest, add other answer option "Can't measure it". The question about whether tall trucks fit through is asked

Hm, it seems to me that it just moves the issue about that it is problematic to estimate whether something is above or below 4.5 in height one quest later. After all, it is perceivable that the user is not able to measure it for other reasons than it being so tall that measuring with AR does not work properly.
(Also, it does not seem to solve the issue for Israel, as they don't want to see `default` anywhere.)

**Comment by mnalis** [comment]
> Hm, it seems to me that it just moves the issue about that it is problematic to estimate whether something is above or below 4.5 in height one quest later

**Comment by mnalis** [comment]
> Hm, it seems to me that it just moves the issue about that it is problematic to estimate whether something is above or below 4.5 in height one quest later

I think the main benefit would be that **`AddMaxHeight` quest becomes easy to solve and not stressful** for newbie users (i.e. when only answers are copying number from sign, or _"there is no sign"_ there is no thinking required), which would allow them to be solved easily by any user (and it would **also sidestep Israel issue** as a bonus). Even if nobody follows with `AddMaxPhysicalHeight` quest it would still (at least partially, i.e. `maxheight:signed=no`) solve them.

Thus only `AddMaxPhysicalHeight` quest would require user expertise and/or extra willingness to engage in significantly more difficult quest (i.e.  AR measuring, or guesstimating the max height). So I would guess only users who really care about categorizing of non-signed heights (**and** are willing to invest extra effort into it) would keep that quest enabled. 

(For example in my case, even with AR available, I have only enabled AR quest for cycleways because I care about bicycle infrastructure **a lot**, and even then it is only present in 20% of my presets. All other quests which require AR/meter are only present in "Everything" preset that I keep for rare moments when I am really bored in almost-100%-solved areas).

**Comment by westnordost** [comment]
I see, well, I see that rather as an argument to completely remove it, because also the `AddMaxPhysicalHeight` quest is not supposed to be reserved for experts (none is).

I'll remove it, then.

**Comment by matkoniecz** [comment]
And sorry to @yrtimiD and @zstadler for bothering you for so long in discussion - but as seen the same applies elsewhere, not only in Israel - so removing it in Israel only would not be the optimal solution.

**Comment by zstadler** [comment]
The initial request was for Israel only because I was not familiar with the situation elsewhere.
I hope that this discussion will lead to a broader solution that will also address the situation in Israel.

**Comment by matkoniecz** [comment]
It happened already, see https://github.com/streetcomplete/StreetComplete/commit/2c1493b639786941772896d00210ac5b38641817 (just commited, not yet released)
